lib/util/source-code-util.js

```diff
@@ -0,0 +1,95 @@
+//------------------------------------------------------------------------------
+
+var assign = require("object-assign"),
+    CLIEngine = require("../cli-engine"),
+    eslint = require("../eslint"),
+    globUtil = require("./glob-util"),
+    defaultOptions = require("../../conf/cli-options");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get the SourceCode object for a single file
+ * @param   {string}     filename The fully resolved filename to get SourceCode from.
+ * @param   {Object}     options  A CLIEngine options object.
+ * @returns {SourceCode}          The SourceCode object representing the file.
+ */
+function getSourceCodeOfFile(filename, options) {
+    var opts = assign({}, options, { rules: {}, baseConfig: {
+        env: { es6: true, commonjs: true, node: true, browser: true },
+        parser: 'babel-eslint',
+        parserOptions: { ecmaVersion: 6, sourceType: "module", ecmaFeatures: { jsx: true, experimentalObjectRestSpread: true } }
+    }});
+    // var opts = assign({}, options, { rules: {}, parser: 'babel-eslint' });
+    var cli = new CLIEngine(opts);
+    cli.executeOnFiles([filename]);
+    var sourceCode = eslint.getSourceCode();
+    // console.log(require('util').inspect(sourceCode.ast.body, { depth: null }));
+    return sourceCode;
+}
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+
+/**
+ * This callback is used to measure execution status in a progress bar
+ * @callback progressCallback
+ * @param {number} The total number of times the callback will be called.
+ */
+
+/**
+ * Gets the SourceCode of a single file, or set of files.
+ * @param   {string[]|string}  patterns   A filename, directory name, or glob,
+ *                                        or an array of them
+ * @param   {Object}           [options]  A CLIEngine options object. If not provided,
+ *                                        the default cli options will be used.
+ * @param   {progressCallback} [cb]       Callback for reporting execution status
+ * @returns {Object}                      The SourceCode of all processed files.
+ */
+function getSourceCodeOfFiles(patterns, options, cb) {
+    var sourceCodes = {},
+        filenames;
+
+    if (typeof patterns === "string") {
+        patterns = [patterns];
+    }
+
+    if (typeof options === "undefined") {
+        options = defaultOptions;
+    } else if (typeof options === "function") {
+        cb = options;
+    }
+
+    patterns = globUtil.resolveFileGlobPatterns(patterns, options.extensions);
+    filenames = globUtil.listFilesToProcess(patterns, options);
+    filenames.forEach(function(filename) {
+        var sourceCode = getSourceCodeOfFile(filename, options);
+        if (sourceCode) {
```

Comment:
Yes, I believe so.  There are times that `getSourceCodeOfFile` does not return any `sourceCode`, particularly if there are parsing errors.  

Whoops, looks like I have an errant console.log commented out on the line beneath...